quickstarts/deploy-vm/main.go

```diff
@@ -0,0 +1,159 @@
+	}
+}
+
+func main() {
+	group, err := createGroup()
+	if err != nil {
+		log.Fatalf("failed to create group: %v", err)
+	}
+	log.Printf("created group: %v\n", *group.Name)
+
+	log.Printf("starting deployment\n")
+	result, err := createDeployment()
+	if err != nil {
+		log.Fatalf("Failed to deploy correctly: %v", err)
+	}
+	log.Printf("Completed deployment: %v", *result.Name)
+	getLogin()
+}
+
+// Create a resource group for the deployment.
+func createGroup() (resources.Group, error) {
+	groupsClient := resources.NewGroupsClient(config.SubscriptionID)
+	groupsClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	return groupsClient.CreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		resources.Group{
+			Location: to.StringPtr(resourceGroupLocation)})
+}
+
+// Create the deployment
+func createDeployment() (resources.DeploymentExtended, error) {
+	template, err := readJSON(templateFile)
+	if err != nil {
+		return resources.DeploymentExtended{}, err
+	}
+	params, err := readJSON(parametersFile)
+	if err != nil {
+		return resources.DeploymentExtended{}, err
+	}
+
+	deploymentsClient := resources.NewDeploymentsClient(config.SubscriptionID)
+	deploymentsClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	deploymentFuture, err := deploymentsClient.CreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		deploymentName,
+		resources.Deployment{
+			Properties: &resources.DeploymentProperties{
+				Template:   template,
+				Parameters: params,
+				Mode:       resources.Incremental,
+			},
+		},
+	)
+	if err != nil {
+		log.Fatalf("Failed to create deployment: %v", err)
+	}
+	err = deploymentFuture.Future.WaitForCompletion(ctx, deploymentsClient.BaseClient.Client)
+	if err != nil {
+		log.Fatalf("Error while waiting for deployment creation: %v", err)
+	}
+	return deploymentFuture.Result(deploymentsClient)
+}
+
+// Get login information by querying the deployed public IP resource.
+func getLogin() {
+	params, err := readJSON(parametersFile)
+	if err != nil {
+		log.Fatalf("Unable to read parameters. Get login information with `az network public-ip list -g %s", resourceGroupName)
+	}
+
+	addressClient := network.NewPublicIPAddressesClient(config.SubscriptionID)
+	addressClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	ipName := (*params)["publicIPAddresses_QuickstartVM_ip_name"].(map[string]interface{})
+	ipAddress, err := addressClient.Get(ctx, resourceGroupName, ipName["value"].(string), "")
+	if err != nil {
+		log.Fatalf("Unable to get IP information. Try using `az network public-ip list -g %s", resourceGroupName)
+	}
+
+	vmUser := (*params)["vm_user"].(map[string]interface{})
+	vmPass := (*params)["vm_password"].(map[string]interface{})
+
+	log.Printf("Log in with ssh: %s@%s, password: %s",
+		vmUser["value"].(string),
+		*ipAddress.PublicIPAddressPropertiesFormat.IPAddress,
+		vmPass["value"].(string))
+}
+
+func readJSON(path string) (*map[string]interface{}, error) {
+	_json, err := ioutil.ReadFile(path)
```

Comment:
>_ [](start = 1, length = 1)

Underscores are not the most idiomatic go around

Reply:
✅